packages/stark-ui/src/modules/generic-search/classes/abstract-search-component.ts

```diff
@@ -0,0 +1,196 @@
+import { ReplaySubject, Subscription } from "rxjs";
+import { map, take } from "rxjs/operators";
+import { OnDestroy, OnInit } from "@angular/core";
+
+import { AbstractStarkFormComponent } from "../classes/abstract-form-component";
+import { StarkGenericSearchService } from "../classes/generic-search.service.intf";
+import { StarkSearchState } from "../entities/search-state.entity.intf";
+import { StarkErrorImpl, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+import { StarkFormUtil } from "../../../util/form";
+import { FormGroup } from "@angular/forms";
+import { StarkProgressIndicatorConfig, StarkProgressIndicatorService, StarkProgressIndicatorType } from "../../progress-indicator";
+
+/**
+ * Default progress indicator configuration
+ */
+const defaultProgressIndicatorConfig: StarkProgressIndicatorConfig = {
+	topic: "",
+	type: StarkProgressIndicatorType.SPINNER
+};
+
+export abstract class AbstractStarkSearchComponent<SearchResultsType, CriteriaType> extends AbstractStarkFormComponent<CriteriaType>
+	implements OnInit, OnDestroy {
+	/** @internal */
+	private _latestResults: Readonly<SearchResultsType[]>;
+	private searchStateSubscription: Subscription;
+
+	/**
+	 * Whether a new search should be performed automatically after initialization in case the last search criteria can be fetched
+	 * from the application state (@ngrx/store)
+	 */
+	protected performSearchOnInit: boolean;
+	/**
+	 * Whether the latest emitted results by the emitResult() method will be kept in the latestResults variable
+	 */
+	protected preserveLatestResults: boolean;
+	/**
+	 * The config of the progress indicator to be shown/hidden when performing the search.
+	 */
+	protected progressIndicatorConfig: StarkProgressIndicatorConfig;
```

Comment:
Travis build is failing now because this property is `protected` here and it is used in the `demo-generic-search-page.component.html`... so ngc throws an exception